catchup/catchpointService.go

```diff
@@ -161,9 +161,12 @@ func MakeNewCatchpointCatchupService(catchpoint string, node CatchpointCatchupNo
 
 // Start starts the catchpoint catchup service ( continue in the process )
 func (cs *CatchpointCatchupService) Start(ctx context.Context) error {
-	err := cs.checkLedgerDownload()
-	if err != nil {
-		return fmt.Errorf("aborting catchup Start(): %s", err)
+	// Only check catchpoint ledger validity if we're starting new
+	if cs.stage == ledger.CatchpointCatchupStateInactive {
```

Comment:
still... why not in MakeService?

Reply:
It feels like bad practice to start making network requests during construction. We really care about whether or not we'll be able to fetch the ledger when the service is about to start.

It's not a big deal to me--I'm ok moving it to MakeService if needed.

Reply:
This is valid point so yes, Start fits better.
The last concert I have is about the following scenarios:
1. at the moment there are 100+ relays and we only try 10
2. what if for some reason the catchpoint requested is found in Start but not found when download + new peer selector invoked?
I think it would be nice to save a peer that 100% has this catchpoint and use this peer as a first one the download loop.